File: src/main/java/com/example/desafio/api/adviceControllerExeception/ControllerExceptionsAdvices.java
package com.example.desafio.api.adviceControllerExeception;

import com.example.desafio.api.exceptions.GetAllCarsException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class ControllerExceptionsAdvices {

    @ExceptionHandler(GetAllCarsException.class)
    public ResponseEntity<String> getAllCarsExceptionHandler(GetAllCarsException e) {
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
    }
}

File: src/main/kotlin/com/example/desafio/api/adviceControllerExeception/ControllerExceptionsAdvices.kt
package com.example.desafio.api.adviceControllerExeception

import com.example.desafio.api.exceptions.GetAllCarsException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class ControllerExceptionsAdvices {
    @ExceptionHandler(GetAllCarsException::class)
    fun getAllCarsExceptionHandler(e: GetAllCarsException): ResponseEntity<String?> {
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.message)
    }
}